unidist/core/backends/mpi/core/communication.py

```diff
@@ -380,38 +471,211 @@ def recv_complex_operation(comm, source_rank):
     -------
     object
         Received data object from another MPI process.
+
+    Notes
+    -----
+    De-serialization is a simple pickle.load in this case
     """
-    return recv_complex_data(comm, source_rank)
+    return comm.recv(source=source_rank)
 
 
-def send_simple_operation(comm, operation_type, operation_data, dest_rank):
+def send_operation_data(comm, operation_data, dest_rank, is_serialized=False):
     """
-    Send operation and Python standard data types.
+    Send data that consists of different user provided complex types, lambdas and buffers.
+
+    The data is serialized with ``unidist.core.backends.mpi.core.ComplexDataSerializer``.
+    Function works with already serialized data.
 
     Parameters
     ----------
     comm : object
         MPI communicator object.
-    operation_type : unidist.core.backends.mpi.core.common.Operation
+    operation_data : object
+        Data object to send.
+    dest_rank : int
+        Target MPI process to transfer data.
+    is_serialized : bool
+        `operation_data` is already serialized or not.
+
+    Returns
+    -------
+    dict or None
+        Serialization data for caching purpose or nothing.
+
+    Notes
+    -----
+    Function returns ``None`` if `operation_data` is already serialized,
+    and the ``dict`` in case the data was serialized and packed in a dictionary.
```

Comment:
```suggestion
    ``None`` is returned if `operation_data` is already serialized,
    otherwise ``dict`` containing data serialized in this function.
```